pallets/parachain-staking/src/lib.rs

```diff
@@ -1522,11 +1624,42 @@ pub mod pallet {
 				let total_paid = pct_due * payout_info.total_staking_reward;
 				let mut amt_due = total_paid;
 				// Take the snapshot of block author and delegations
-				let state = <AtStake<T>>::take(paid_for_round, &collator);
+
+				// Decode [CollatorSnapshot] depending upon when the storage was migrated
+				let is_at_stake_migrated = <MigratedAtStake<T>>::get()
+					.map_or(false, |migrated_at_round| {
+						paid_for_round >= migrated_at_round
```

Comment:
why do we do this check?

Reply:
We changed the `CollatorSnapshot::delegations` field to a new type and instead of doing a migration, we will do a on-the-fly-migration. Basically when a new `CollatorSnapshot` is taken, `MigratedAtStake` is updated to that round. This code checks if that storage exists (or not) with the current round to decide which storage to decode.

Reply:
ah I see, thanks!